server/src/test/java/io/crate/expression/scalar/ArraySliceFunctionTest.java

```diff
@@ -0,0 +1,193 @@
+                                        "an array slice ([<from>:<to>])");
+        assertEvaluate("'not an array'[1:3]", null);
+    }
+
+    @Test
+    public void testFromIsNegative() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][-1:]", null);
+    }
+
+    @Test
+    public void testToIsNegative() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][:-1]", null);
+    }
+
+    @Test
+    public void testFromIsBig() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][2147483648:]", List.of());
+    }
+
+    @Test
+    public void testToIsBig() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][:2147483648]", List.of(1, 2, 3, 4, 5));
+    }
+
+    @Test
+    public void testFromIsBigExpression() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][2147483647+20:]", List.of());
+    }
+
+    @Test
+    public void testToIsBigExpression() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][:2147483647+20]", List.of(1, 2, 3, 4, 5));
+    }
+
+    @Test
+    public void testFromIsZero() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][0:]", List.of());
+    }
+
+    @Test
+    public void testToIsZero() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][:0]", List.of());
+    }
+
+    @Test
+    public void testFromIsZeroExpression() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][10-10:]", List.of());
+    }
+
+    @Test
+    public void testToIsZeroExpression() {
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("Array index must be in range 1 to 2147483647");
+        assertEvaluate("[1,2,3,4,5][:10-10]", List.of());
+    }
+
```

Comment:
Maybe worth adding a test case using the scalar name? like `array_slice([1,2,3], 1, 2)`.
Would be nice also, to document the `array_slice()` scalar additionally to the the array access documentation.